File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/VB.NET/7300/D7300_DI_Single_Buffer_Sync/D7300_DI_Single_Buffer_Sync/Form1.vb
Public Class Form1
    Dim m_dev As Short
    Dim data_buffer(99) As UShort

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        m_dev = Register_Card(PCI_7300A_RevB, 0)
        If (m_dev < 0) Then
            MessageBox.Show("DO_7300B_Config error!")
        End If
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        Dim ret As Short
        If (m_dev >= 0) Then
            ret = DASK.Release_Card(m_dev)
        End If
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim i As Integer
        Dim err As Short
        err = DASK.DI_7300B_Config(m_dev, 16, TRIG_INT_PACER, P7300_WAIT_NO, P7300_TERM_OFF, P7300_DIREQ_POS Or P7300_DIACK_POS Or P7300_DITRIG_POS, False, False)
        If (err <> 0) Then
            MessageBox.Show("DI_7300B_Config error!")
            Return
        End If
        err = DI_ContReadPort(m_dev, 0, data_buffer(0), 100, 10000, SYNCH_OP)
        If (err <> 0) Then
            MessageBox.Show("DI_ContReadPort error!")
            Return
        End If
        listBox1.Items.Clear()
        For i = 0 To 99
            listBox1.Items.Add(data_buffer(i))
        Next
    End Sub
End Class

File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/C#/7300/D7300_DI_Single_buffer_Sync/D7300_DI_Single_buffer_Sync/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace D7300_DI_Single_buffer_Sync
{
    public partial class Form1 : Form
    {
        ushort[] data_buffer = new ushort[100];
        public short m_dev;
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            m_dev = DASK.Register_Card(DASK.PCI_7300A_RevB, 0);
            if (m_dev < 0)
            {
                MessageBox.Show("Register_Card error!");

            }
        }

        private void Form1_InputLanguageChanged(object sender, InputLanguageChangedEventArgs e)
        {
            short ret;
            if (m_dev >= 0)
            {
                ret = DASK.Release_Card((ushort)m_dev);
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            int i;
            short err;
            err = DASK.DI_7300B_Config((ushort)m_dev, 16, DASK.TRIG_INT_PACER, DASK.P7300_WAIT_NO, DASK.P7300_TERM_OFF, DASK.P7300_DIREQ_POS | DASK.P7300_DIACK_POS | DASK.P7300_DITRIG_POS, false, false);
            if (err != 0)
            {
                MessageBox.Show("DI_7300B_Config error!");
                return;
            }
            err = DASK.DI_ContReadPort((ushort)m_dev, 0, data_buffer, 100, 10000, DASK.SYNCH_OP);
            if (err != 0)
            {
                MessageBox.Show("DI_ContReadPort error!");
                return;
            }
            listBox1.Items.Clear();
            for (i = 0; i < 100; i++)
            {
                listBox1.Items.Add(data_buffer[i]);
            }
        }
    }
}